src/components/progress-tracker/progress-tracker.component.tsx

```diff
@@ -75,40 +54,12 @@ const ProgressTracker = ({
   maxProgressLabel,
   labelsPosition = "top",
   labelWidth,
+  screenReaderOnlyLabel,
   ...rest
 }: ProgressTrackerProps) => {
-  if (
-    (ariaLabel ||
-      ariaDescribedBy ||
-      ariaValueNow ||
-      ariaValueMax ||
-      ariaValueMin ||
-      ariaValueText) &&
-    !deprecatedAriaTagsWarnTriggered
-  ) {
-    deprecatedAriaTagsWarnTriggered = true;
-    Logger.deprecate(
-      "The 'aria-' attribute props in `ProgressTracker` have been deprecated and will soon be removed."
-    );
-  }
-
   const l = useLocale();
-  const barRef = useRef<HTMLDivElement>(null);
-  const [barLength, setBarLength] = useState("0px");
   const prefixLabels = labelsPosition !== "bottom";
 
-  const updateBarLength = useCallback(() => {
-    setBarLength(`${barRef.current?.offsetWidth}px`);
-  }, []);
-
-  useLayoutEffect(() => {
-    updateBarLength();
-  }, [updateBarLength]);
-
-  useResizeObserver(barRef, () => {
-    updateBarLength();
-  });
-
```

Comment:
**praise**: the new CSS approach of ensuring the inner bar length is proportional to the progress value is much simplier. Good spot!